src/csharp/Grpc.Core/Internal/GrpcThreadPool.cs

```diff
@@ -82,7 +85,8 @@ public void Start()
 
                 for (int i = 0; i < poolSize; i++)
                 {
-                    threads.Add(CreateAndStartThread(i));
+                    var optionalProfiler = i < threadProfilers.Count ? threadProfilers[0] : null;
```

Comment:
The idea is that if you want to enable profiling, you only need to edit one line of code by adding a profiler to the list of profilers (and usually it's enough to add profiler just for the first thread - other thread yield similar results and processing the dump doesn't take that much time.

I am aware this can be considered a bit dirty, but currently it's the easiest way to achieve what I need and when disabled it has no performance/memory impact (and the complexity increase is negligible). Please let me know what your opinion is.